### ISSUE TITLE
Dependency on CURL_OPENSSL_3

### ISSUE THREAD

**Issue by sudara** [Tracktion/pluginval#44]
Firstly: Thanks so much for this tool! It's amazing and I'm looking forward to running it in CI.

Not sure if this is known or even a pluginval (vs. JUCE) thing, but similar to #3, out of the box ubuntu 20.04.2 (via github actions) [chokes on running the latest pluginval](https://github.com/sudara/pamplejuce/runs/2438921211?check_suite_focus=true) with:

```
./pluginval: /lib/x86_64-linux-gnu/libcurl.so.4: version `CURL_OPENSSL_3' not found (required by ./pluginval)
```

It looks like [https://github.com/Tracktion/pluginval/blob/develop/CMakeLists.txt#L73-L74](JUCE_WEB_BROWSER and JUCE_USE_CURL) are set to 0 in pluginval, so thought i'd bubble it up.

**Comment by drowaudio** [comment]
That's odd, I would have thought with those set to 0 it wouldn't have any dependency on CURL?
It might be this was a JUCE bug with the cmake implementation that has been fixed. I think pluginval will need a few changes to update JUCE though.

**Comment by sudara** [comment]
No prob, just wanted to bubble it up!

**Comment by sudara** [comment]
> I think pluginval will need a few changes to update JUCE though.

Felt motivated, so I updated pluginval to latest JUCE develop locally and everything seemed happy running against my plugin. Building cmake on the command line (xcode generator) also seemed happy... 

Happy to dig in further if there are known issues that need attention!

**Comment by drowaudio** [comment]
Ah ok, thats good to hear. I'll see if I can update soon. I might wait until the next stable JUCE release though as it's a more logical point to do it.

**Comment by sudara** [comment]
I looked into running tracktion on JUCE 6.1, it looks like there are a couple spots affected by the [modal loop changes](https://github.com/juce-framework/JUCE/commit/fe4ba9071be037bbed58268e1f4139f1dd23e15c#diff-d9fb8759cf97ab5a462e8afdada00280babf71dd5eb7d7e2708f932352c099a0). I could open a PR that moves those to async or sets `JUCE_MODAL_LOOPS_PERMITTED=1` if you let me know which you prefer.

**Comment by drowaudio** [comment]
This should be sorted on the develop branch: https://github.com/Tracktion/tracktion_engine/tree/develop

**Comment by sudara** [comment]
Whups, I don’t know why I said tracktion, I meant pluginval, this repo….

**Comment by drowaudio** [comment]
Yeah, at the moment you'll have to use the version of JUCE that is provided as a git submodule.
I will update this at some point in the future though. If you want to do that and provide a PR that could help. Thanks.
